manifests/server/service.pp

```diff
@@ -47,4 +47,18 @@
     File['mysql-config-file'] -> Service['mysqld']
   }
 
+  if $mysql::server::override_options and $mysql::server::override_options['mysqld'] and $mysql::server::override_options['mysqld']['socket'] {
+    $mysqlsocket = $mysql::server::override_options['mysqld']['socket']
+  } else {
+    $mysqlsocket = $options['mysqld']['socket']
+  }
+
+  exec { 'wait_for_mysql_socket_to_open':
+    command     => "ls ${mysqlsocket}",
+    tries       => '3',
+    try_sleep   => '10',
+    subscribe   => Service['mysqld'],
+    refreshonly => true,
```

Comment:
you *could* use `git push --force`
however, you've been working on your own master branch, which might make things very confusing, especially in case you have to [rebase](https://www.git-scm.com/book/en/v2/Git-Branching-Rebasing), etc…

i highly recommend using the [github flow](https://guides.github.com/introduction/flow/index.html) work flow.